src/lang.rs

```diff
@@ -845,20 +845,25 @@ impl Context for CallStack {
         let mut env = self.env();
         loop {
             // search in this environment for value by name
-            if let Some(value) = env.values.borrow().get(&name) {
-                let result = value.clone();
-                return match result {
-                    c @ Obj::Closure(..) => c.force(self),
-                    _ => Ok(result),
-                };
-            }
-
-            // if not found, search through parent if available
-            if let Some(parent) = &env.parent {
-                env = parent.clone();
-            } else {
-                break;
-            }
+            let Some(value) = env.values.borrow_mut().get(&name).cloned() else {
+                // if not found, search through parent if available
+                if let Some(parent) = &env.parent {
+                    env = parent.clone();
+                    continue;
+                } else {
+                    break;
+                }
+            };
```

Comment:
@sebffischer emphasizing for awareness since you're thinking critically about the reference counting and mutability of objects.

This is how values are retrieved from an environment. Note that currently all values are _always `.cloned()`_ when they're accessed in an environment. This means that doing something as simple as:

```r
a <- 1
b <- a
c <- b
```

Produces three copies of the same data, despite no mutations. This is a known issue, just something I haven't addressed yet. 

I always imagined objects would need some sort of global reference counting at some point. Perhaps what is now `Obj` should likewise be converted into `ObjType` and we should have a similar pattern such as `Obj<Rc<ObjType<...>>>`.

This starts to look a tad redundant with the current vector stuff. 

Frankly, I think it's good to get the vectors sorted out in isolation. There's enough complexity there to really stress-test a concept. At some point we'll have to circle back and develop that idea into something more universal.